File: Builder/Ruby/Report.rb
class Report
    attr_writer :filename, :content

    def describe
        if @filename
            puts "The name of file is #{@filename}"
        else
            puts "The file doesn't has any name"
        end

        if @content
            puts "The content of the file is #{@content}"
        else
            puts "The file doesn't has any content"
        end
    end
end

File: Builder/Python 3/Report.py
class Report:
  def __init__(self):
    self.filename = None
    self.content = None

  def describe(self):
    if self.filename:
      print("The name of file is {}".format(self.filename))
    else:
      print("The file doesn't have a name")
    
    if self.content:
      print("The content of file is {}".format(self.content))
    else:
      print("The file doesn't have a content")